server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -128,6 +128,10 @@ public static boolean isSearchNode(Settings settings) {
         return hasRole(settings, DiscoveryNodeRole.SEARCH_ROLE);
     }
 
+    public static boolean isOfflineNode(Settings settings) {
+        return hasRole(settings, DiscoveryNodeRole.OFFLINE_ROLE);
+    }
+
```

Comment:
`DiscoveryNode` has instance methods for these too. Do we need one for the offline role?